examples/dctool_download.c

```diff
@@ -96,7 +96,7 @@ dive_cb (const unsigned char *data, unsigned int size, const unsigned char *fing
 
 cleanup:
 	dc_parser_destroy (parser);
-	return 1;
+	return rc == DC_STATUS_SUCCESS;
```

Comment:
IMO (and also in others' opinions [0]) this is the correct behavior.

Especially while debugging I want to stop at the first error and not continue to do stuff that I most likely will fail at again. It's not that you're magically recovering from a DC sending an unknown format (or the parser not working correctly (yet)).

And I believe that's also the case when "only" running the tools "as user".

You're right that if it's a one-off error, you'll not be able to continue reading other values, but I'm not sure if you should trust the other values you can read from the DC in that case.

[0] https://www.martinfowler.com/ieeeSoftware/failFast.pdf

Reply:
> But if the user can't download the dives, I also can't fix the problem.

Shouldn't the `RAW` download mode be the one that you get from the user? The logs would be the same, but there's no parsing happening... 

Would an option `-s, --strict               Enable strict parsing (Exit if there has been an error while parsing)` be acceptable?

Reply:
The RAW output is not (always) a solution here. When people encounter a problem, the easiest solution is to simply send the log of the failed download, without having to repeat the download again. Some dive computers are painfully slow, so unless really necessary you don't want to repeat a 10 or 30 minute download to get the same data again. And the RAW output mode does not necessary fix the problem either. The parser is still needed in RAW mode for the template parameters. Thus if the dc_parser_new() function fails, the download can still be aborted.

I'm not really sure about the advantage of such a strict option? Maybe a better alternative is to keep track of parsing errors and let the application return EXIT_FAILURE after the download has finished to indicate something went wrong? You can still download as much dives as possible, but you also get an indication something went wrong.

Reply:
> I'm not really sure about the advantage of such a strict option?

While implementing this I found it very useful if the download stopped after the first failure instead of me having to terminate the process manually each time.

> Maybe a better alternative is to keep track of parsing errors and let the application return EXIT_FAILURE after the download has finished to indicate something went wrong?

That could be done additionally.